discovery/service.go

```diff
@@ -72,7 +101,7 @@ func (service *ServiceDefinition) registerService() error {
 			EnableTagOverride: service.EnableTagOverride,
 			Check: &api.AgentServiceCheck{
```

Comment:
After looking up that this is using Consul's `api.AgentServiceCheck` I believe the configuration option should be on the job level and _not_ within the health check. Optionally, we could expose the same `initial_status` option for all health checks independently... but I think that should be a future enhancement warranted from user requests for the feature.